code/game/objects/structures/broken_flooring.dm

```diff
@@ -0,0 +1,54 @@
+
+/obj/structure/broken_flooring
+	name = "broken tiling"
+	desc = "A segment of broken flooring."
+	icon = 'icons/obj/brokentiling.dmi'
+	icon_state = "corner"
+	anchored = TRUE
+	density = FALSE
+	opacity = FALSE
+	plane = FLOOR_PLANE
+	layer = CATWALK_LAYER
+
+/obj/structure/broken_flooring/Initialize(mapload)
+	. = ..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/structure/broken_flooring/LateInitialize()
+	. = ..()
+	var/turf/turf = get_turf(src)
+	if(!istype(turf, /turf/open/floor/plating)) // Render as trash above the current tile
+		plane = GAME_PLANE
+		layer = LOW_OBJ_LAYER
+		return
+	for(var/obj/O in turf)
```

Comment:
no new single letter variables, if it's really hot then you should add a `CHECK_TICK` somewhere so we dont' choke the server out